src/metabase/sync/analyze/classifiers/domain_entity.clj

```diff
@@ -0,0 +1,24 @@
+(ns metabase.sync.analyze.classifiers.domain-entity
+  (:require [metabase.domain-entities.core :as de]
+            [metabase.sync.interface :as i]
+            [metabase.util :as u]
+            [schema.core :as s]
+            [toucan.db :as db]))
+
+(s/defn assign-domain-entity! :- i/TableInstance
+  "Find and instantiate the best fitting domain model."
+  [table :- i/TableInstance]
+  (let [domain-entity (de/domain-entity-for-table table)]
+    (assoc table :domain_entity (cond
+                                  (nil? domain-entity)
+                                  nil
+
+                                  (:domain_entity table)
+                                  (do
+                                    (db/update! 'DomainEntity (:domain_entity table) domain-entity)
```

Comment:
same thing as other PR, you shouldn't quote the Toucan entities unless it's done because doing otherwise would create a circular reference. That way when running with `lein ring server` the namespace graph is correct